docs/ruleset_functions/get_hvac_sys_and_assoc_zones_largest_exhaust_source.md

```diff
@@ -0,0 +1,41 @@
+
+**Description:** Returns a list with the sum of the hvac fan system exhaust fan cfm values, the maximum zone level exhaust source across the zones associated with the HVAC system, the number of exhaust fans associated with the hvac fan system, and the maximum cfm of all of the exhaust fans associated with the hvac system fan system [hvac_sys_exhaust_cfm_sum, maximum_zone_exhaust, num_hvac_exhaust_fans, maximum_hvac_exhaust]. This is for evaluating G3.1.2.10 exception 6 which states "Where the largest exhaust source is less than 75% of the design outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed design. "
+
+**Inputs:**  
+- **U,P,or B-RMI**: The RMI in which the largest exhaust source CFM is to be returned.  
+- **hvac_sys**: The hvac system object for which the largest exhaust source is being determined.
+
+**Returns:**  
+- **get_hvac_sys_and_assoc_zones_largest_exhaust_source**: Returns a list with the sum of the hvac fan system exhaust fan cfm values, the maximum zone level exhaust source across the zones associated with the HVAC system, the number of exhaust fans associated with the hvac fan system, and the maximum cfm of all of the exhaust fans associated with the hvac system fan system [hvac_sys_exhaust_cfm_sum, maximum_zone_exhaust, num_hvac_exhaust_fans, maximum_hvac_exhaust]. This is for evaluating G3.1.2.10 exception 6 which states "Where the largest exhaust source is less than 75% of the design outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed design. "  
+ 
+**Function Call:**   
+1. get_dict_of_zones_and_terminal_units_served_by_hvac_sys()   
+
+
+## Logic:  
+- Create dictionary of hvac systems, zones, and terminal units: `dict_of_zones_and_terminal_units_served_by_hvac_sys = get_dict_of_zones_and_terminal_units_served_by_hvac_sys(B_RMI)`  
+- Get the list of zones associated with the hvac system: `zone_list = dict_of_zones_and_terminal_units_served_by_hvac_sys[hvac_sys.id]["ZONE_LIST"]`  
+- Set largest zone exhaust source cfm equal to zero: `maximum_zone_exhaust = 0`  
+- Set largest hvac exhaust cfm equal to zero: `maximum_hvac_exhaust = 0`  
+- Create a fan system object: `fan_system_p = sys.fan_system`   
+- Get list of exhaust fan fan objects: `fan_sys_exhaust_list = list(fan_system_p.exhaust_fans)`  
+- Get the number of exhaust fans associated with the hvac system: `num_hvac_exhaust_fans = len(fan_sys_exhaust_list)`  
+- Reset the total fan cfm variable: `total_fan_cfm = 0` 
+- Reset the fan cfm variable: `fan_cfm = 0` 
+- Check if the num_hvac_exhaust_fans is greater than 0: `if num_hvac_exhaust_fans > 0:`  
+    - Cycle through the list of exhaust fans: `for fan_p in fan_sys_exhaust_list:`  
+        - Get the fan cfm: `fan_cfm = fan_p.design_airflow`  
+        - Total the fan cfm across the multiple fans: `total_fan_cfm = total_fan_cfm + fan_cfm`  
```

Comment:
Why do we need the total? It was included in my original comment, but you questioned it in the email and I think we agreed that you were right and we don't need it? The energy recovery exception refers to a single fan accounting for 75% of the total design outdoor air flow, not the total design exhaust flow.

Reply:
What if there was an array of exhaust fans? I think the way this function and 19-24 is setup works. 

Basically what it does is it checks whether the max of the zone level exhaust and the sum of the exhaust associated with the hvac system is less 75% of the OA cfm. If yes then the exception applies. I think this makes sense because if there is more than 1 exhaust fan associated with the hvac system and it sums to less than 75% of the OA cfm then this exception would apply still. If there is only 1 exhaust fan associated with the hvac system then the sum would equal the cfm of the exhaust fan which is what we are interested in comparing.

The RDS then looks at the scenario where there is more than 1 exhaust fan associated with the HVAC system. It checks whether the max of the max of the zone level exhaust and the max exhaust cfm associated with the hvac system exhaust fans is less 75% of the OA cfm. If yes then it returns an UNDETERMINED outcome explaining that the exception may apply. I think this makes sense. Am I missing something?